lib/CRandomGenerator.cpp

```diff
@@ -25,47 +26,58 @@ CRandomGenerator::CRandomGenerator(int seed)
 
 void CRandomGenerator::setSeed(int seed)
 {
+	logRng->debug("setSeed(%d)", seed);
 	rand.seed(seed);
 }
 
 void CRandomGenerator::resetSeed()
 {
+	logRng->trace("resetSeed()");
 	boost::hash<std::string> stringHash;
 	auto threadIdHash = stringHash(boost::lexical_cast<std::string>(boost::this_thread::get_id()));
 	setSeed(static_cast<int>(threadIdHash * std::time(nullptr)));
 }
 
 TRandI CRandomGenerator::getIntRange(int lower, int upper)
 {
+	logRng->trace("getIntRange(%d, %d)", lower, upper);
```

Comment:
This logging would likely become very spammy. And we already have a lot of logging.

Reply:
Would you please clarify what are you suggesting here - should I remove the `trace` logging (getIntRange, getDoubleRange, etc.) and leave only the `debug` logging (nextInt, nextDouble, etc.)? Or should the entire RNG log be removed?

Reply:
Not sure. 
At the moment even `trace` level is written to log file. And it is not uncommon for log file size to be measured in hundreds of Mb's.

So we probably should either remove it or change logging level of logger to `debug`. Although don't remember where this is configured.

Reply:
> At the moment even trace level is written to log file

That sounds incorrect to me -- at least it does not correlate to what I am observing locally. Please tell me where am I wrong -- here is what I know:

The log levels are set per-domain in `settings.json` (platform-dependant, on macOS is `~/Library/Application Support/vcmi/config/settings.json`), and here is what I have in mine:

```json
    "loggers" : [
      {"domain" : "global", "level" : "error"},
      {"domain" : "ai", "level" : "error"},
      {"domain" : "rng", "level" : "error"},
      {"domain" : "network", "level" : "trace"},
      {"domain" : "mod", "level" : "error"},
      {"domain" : "animation", "level" : "error"},
      {"domain" : "bonus", "level" : "error"}
    ]
```

If a domain is missing from this file, it will fallback to the default value (`trace`) as defined in [config/schemas/settings.json](https://github.com/vcmi/vcmi/blob/1.5.3/config/schemas/settings.json#L540).

One reason I can think of is: new installations come with no `config.json` so they always use the default level (`trace`) for _all_ loggers. Is that the issue?

Reply:
> One reason I can think of is: new installations come with no config.json so they always use the default level (trace) for all loggers. Is that the issue?

Yep. You can configure it that way. But at the moment all players that have not deliberately configured log system (so >99% of them) have all logging domains at 'trace' level

Reply:
@IvanSavenko Should we change it? Trace is definitely not that level needed by default?

Reply:
@smanolloff regarding very extensive logging you can go the way of #ifdef. It is better than log level switching if will be used in very high-load place because even preparing log string may take some CPU.

Reply:
Ah, true, I also thought it initially, then I did some benchmarks and it turned out the performance impact is around 2%, which I would consider negligible (and comparable to the measurement  accuracy error itself). But I agree that  using a preprocessor macro is fine, if it wouldn't degrade readability.
We took the discussion off the GH comment section, but in summary decided to leave this PR open until Ivan decides which changes are worth integrating into his own work branch.

Reply:
> @IvanSavenko Should we change it? Trace is definitely not that level needed by default?

Don't know. It is certainly useful since quite a lot of player reports are simple "game crashed". So detailed info from logs help a lot with tracking down last in-game events.

In theory we could try to disable trace level only for release builds, but release builds is where most player reports come from. If we do so, we should review what we log and decide which trace-level info is useful and should be promoted to 'debug' and what can be left in 'trace' and disabled.

> @smanolloff regarding very extensive logging you can go the way of #ifdef. It is better than log level switching if will be used in very high-load place because even preparing log string may take some CPU.

Actually log string is only generated if log level check passes. Formatting arguments are computed unconditionally though. So log entries like these ones will only generate overhead in form of a single check for log level.

Reply:
If the amount of logged trace messages is a concern, **and** there is no way to turn it off for players by default without affecting all other logging, I would suggest to either:
* think about options for shipping a `settings.json` with pre-configured loglevels (as part of the game installation).
* leave it on for debug builds only, and use it to ensure it's working correctly. For release builds we would only care about the log line `setSeed(), as everything else can be inferred from those. Maybe leaving the logs in the constructor and `resetSeed()` as well for a bit more context, as they are not called often as well.